src/components/Nav/Nav.tsx

```diff
@@ -196,6 +194,14 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     ev.preventDefault();
     ev.stopPropagation();
   }
+
+  private _onNavLinkClicked(link: INavLink, ev: React.MouseEvent): void {
+    this._selectedKey = link.key;
+
+    if (link.onClick) {
+      link.onClick(ev);
```

Comment:
Make sure to test with both button/href cases to ensure unnecessary navigation doesn't lead away the page. ie. preventPropagation etc. But maybe the design is that the click hadler is expected to do that?